pkg/store/cache/inmemory.go

```diff
@@ -289,13 +289,13 @@ func copyToKey(l labels.Label) cacheKeyPostings {
 
 // StorePostings sets the postings identified by the ulid and label to the value v,
 // if the postings already exists in the cache it is not mutated.
-func (c *InMemoryIndexCache) StorePostings(blockID ulid.ULID, l labels.Label, v []byte) {
+func (c *InMemoryIndexCache) StorePostings(blockID ulid.ULID, l labels.Label, v []byte, _ PostingsCodec) {
```

Comment:
I ignored codec for in memory index cache because if using in memory index cache, we don't need to use codec to differentiate the cache key as each store gateway instance is only accessing its own memory so nothing is shared.